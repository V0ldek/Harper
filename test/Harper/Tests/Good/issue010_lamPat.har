value Bag a b c d = {
  data = {
    a :: a; b :: b; c :: c; d :: d;
  }
};

main :: Integer;
main = {
  (extract :: Bag Integer Integer Integer Integer -> Integer) = \Bag { a: (a :: Integer) } => a;
  (bag :: Bag Integer Integer Integer Integer) = val Bag { a = 1, b = 2, c = 3, d = 4 };
  return extract bag;
};