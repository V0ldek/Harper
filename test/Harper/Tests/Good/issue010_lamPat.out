Execution ended with value: 1